plugins/webhooks/index.js

```diff
@@ -166,49 +238,45 @@ function pullRequestClosed(options, request, reply) {
 
 /**
  * Stop any running builds and start the build for the synchronized pull-request
- * @method pullRequestSync
+ * @async  pullRequestSync
  * @param  {Object}       options
  * @param  {String}       options.hookId        Unique ID for this scm event
- * @param  {String}       options.pipelineId    Identifier for the Pipeline
- * @param  {String}       options.name          Name of the job (PR-1)
- * @param  {String}       options.sha           Specific SHA1 commit to start the build with
- * @param  {String}       options.username      User who created the PR
- * @param  {String}       options.scmContext    Scm which pipeline's repository exists in
+ * @param  {String}       options.name          Name of the new job (PR-1)
  * @param  {String}       options.restriction   If we are restricting PRs based on their origin
  * @param  {String}       options.prSource      The origin of this PR
- * @param  {String}       options.prRef         Reference to pull request
  * @param  {Pipeline}     options.pipeline      Pipeline model for the pr
  * @param  {Array}        options.changedFiles  List of files that were changed
  * @param  {Hapi.request} request               Request from user
  * @param  {Hapi.reply}   reply                 Reply to user
  */
-function pullRequestSync(options, request, reply) {
+async function pullRequestSync(options, request, reply) {
     const { pipeline, hookId, restriction, prSource, name, prNum, action } = options;
-    let prJobs;
 
-    // Check for restriction upfront
-    if (isRestrictedPR(restriction, prSource)) {
-        request.log(['webhook', hookId],
-            'Skipping build since pipeline is configured to restrict ' +
-            `${restriction} and PR is ${prSource}`);
+    if (pipeline) {
+        // Check for restriction upfront
+        if (isRestrictedPR(restriction, prSource)) {
+            request.log(['webhook', hookId],
+                'Skipping build since pipeline is configured to restrict ' +
+                `${restriction} and PR is ${prSource}`);
 
-        return reply().code(204);
-    }
+            return reply().code(204);
+        }
 
-    return pipeline.jobs
-        .then((jobs) => {
-            prJobs = jobs.filter(j => j.name.includes(name));
+        pipeline.jobs.then(jobs => jobs.filter(j => j.name.includes(name)))
```

Comment:
The promise object returned is dropped. Is it okay?

Reply:
we don't need returned promise object here, but might need await.
so, I fix to await this.